src/gepa/api.py

```diff
@@ -220,7 +222,13 @@ def optimize(
 
         module_selector = module_selector_cls()
 
-    batch_sampler = EpochShuffledBatchSampler(minibatch_size=reflection_minibatch_size, rng=rng)
+    if batch_sampler is None:
+        batch_sampler = EpochShuffledBatchSampler(minibatch_size=reflection_minibatch_size, rng=rng)
+    else:
+        logger.log(
+            "WARNING: Both 'batch_sampler' and 'reflection_minibatch_size' were provided. "
+            "Ignoring 'reflection_minibatch_size' since 'batch_sampler' takes precedence."
```

Comment:
Let's throw an error instead of warning. Make the corresponding change in the documentation above. I prefer to prevent the user from executing something the developer knows is going to lead to unexpected outcomes.